carpics_360v0.1.js

```diff
@@ -625,24 +787,452 @@ var CarPicsSpinnerAPI = (function() {
+        }
+        /**
+        * Allows drag to scroll image within zoomed element
+        */
+        this.dragMove = function(baseEvent, offset, zoomed) {
+            var parentOffset = this.HTMLElement.parentElement.getBoundingClientRect();
+            var clientX = (baseEvent.type !== "touchmove" ? baseEvent.clientX : baseEvent.targetTouches[0].clientX) - parentOffset.left;
+            var clientY = (baseEvent.type !== "touchmove" ? baseEvent.clientY : baseEvent.targetTouches[0].clientY) - parentOffset.top;
+            this.setVelocity(clientX, clientY);
+        }
+        this.setVelocity = function(Px, Py) {
+            var c = 0.6;    //how much each mouse movement matters 
+            if (typeof this.clientX == "undefined") {
+                this.clientX = Px;
+            }
+            if (typeof this.clientY == "undefined") {
+                this.clientY = Py;
+            }
+            this.velocityX = (Px-this.clientX) * c + this.velocityX;
+            this.velocityY = (Py-this.clientY) * c + this.velocityY;
+            this.clientX = Px;
+            this.clientY = Py;
+            this.moveTimeOut = this.moveTimeOut || setTimeout((function(thisObj) {
+                return function() {
+                    thisObj.inertialMove();
+                }  
+            })(this),10);
+        }
+        this.inertialMove = function () {
+            var decay = 0.8;    // how fast for decay, cannot be greater than 1
+            var threshold = 1;    // the threshold to stop the movement
+            var offset = this.HTMLElement.getBoundingClientRect();
+            var parentOffset = this.HTMLElement.parentElement.getBoundingClientRect();
+            var correctX = offset.left + this.velocityX - parentOffset.left;
+            var correctY =  offset.top + this.velocityY - parentOffset.top;
+            if(correctX>0){
+                correctX=0;
+            } else if (-correctX > (offset.width-parentOffset.width)) {
+                correctX = -(offset.width-parentOffset.width);
+            }
+            if(correctY>0){
+                correctY=0;
+            } else if (-correctY > (offset.height-parentOffset.height)) {
+                correctY = -(offset.height-parentOffset.height);
+            }
+            this.HTMLElement.style.left = correctX + 'px';
+            this.HTMLElement.style.top = correctY + 'px';
+            this.velocityX = this.velocityX * decay;
+            this.velocityY = this.velocityY * decay;
+            if (this.velocityX > threshold || this.velocityY > threshold) {
+                this.moveTimeOut = setTimeout((function(thisObj) {
+                    return function(){
+                        thisObj.inertialMove();
+                    }
+                })(this),10);
             } else {
+                this.moveTimeOut = 0;
+            }
+        }
+        this.listOfPointsOfInterest = [];
+        this.displayPointsOfInterest = function(source){
```

Comment:
Please comment all the magic in this function. (just a line-end comment for each bit of magic)